src/main/java/com/scottlogic/sorts/NameDatePostSorter.java

```diff
@@ -0,0 +1,30 @@
+package com.scottlogic.sorts;
+
+import com.scottlogic.PostSorter;
+import com.scottlogic.SortOrder;
+import com.scottlogic.UserPost;
+
+import java.util.List;
+
+public class NameDatePostSorter implements PostSorter {
+
+    public final DatePostSorter DatePostSorter;
+    public final AuthorPostSorter AuthorPostSorter;
+
+    public NameDatePostSorter(DatePostSorter DatePostSorter, AuthorPostSorter AuthorPostSorter) {
```

Comment:
Here it is probably nicer for the user for this class to construct the extra classes rather than force that burden of work on the person using the constructor. Especially considering that the only real options are the empty constructor sorters